state/factory/workingset.go

```diff
@@ -142,34 +144,39 @@ func withActionCtx(ctx context.Context, selp action.SealedEnvelope) (context.Con
 
 func (ws *workingSet) runAction(
 	ctx context.Context,
-	elp action.SealedEnvelope,
+	selp action.SealedEnvelope,
 ) (*action.Receipt, error) {
 	if protocol.MustGetBlockCtx(ctx).GasLimit < protocol.MustGetActionCtx(ctx).IntrinsicGas {
 		return nil, action.ErrGasLimit
 	}
 	// Reject execution of chainID not equal the node's chainID
-	if !action.IsSystemAction(elp) {
-		if err := validateChainID(ctx, elp.ChainID()); err != nil {
+	if !action.IsSystemAction(selp) {
+		if err := validateChainID(ctx, selp.ChainID()); err != nil {
 			return nil, err
 		}
 	}
+	// for replay tx, check against deployer whitelist
+	g := genesis.MustExtractGenesisContext(ctx)
```

Comment:
This check maybe more suitable to be placed in the `validate` function, executed before the `runAction` function.

Reply:
validate() does not have actions as input, so cannot check the sender address is whitelisted or not